### PR TITLE
Test traits and wrapped contracts using traits

### PR DESCRIPTION
**Pull request by friedger**
This PR is about traits and
* adds a test to call a public function as readonly function
* adds a test to call a public function 
* adds a test to call a readonly function
* adds tests for wrappers of contracts using traits

### CODE REVIEW

**Comment by friedger** [comment]
This PR contains #45

File: tests/integration/traits/trait-parameter.spec.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+      contractName: "test-trait",
+      contractAddress,
+      functionName: "foo-arg",
+      functionArgs: [contractPrincipalCV(contractAddress, "impl-trait")],
+      network,
+      senderAddress: Accounts.WALLET_1.stxAddress,
+    });
+  };
+
+  const callReadOnlyTestTraitCallFo = (
```
Comment (obycode) [inline tests/integration/traits/trait-parameter.spec.ts:79]:
```suggestion
  const callReadOnlyTestTraitCallFoo = (
```
(and the uses below)

File: tests/integration/epoch-2-2/wrap-trait-workaround3.spec.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  const codeBodyImplTrait = `(define-public (foo (a uint))
+(ok a)
+)`;
+
+  const codeBodyTestTraitWrapper = `(define-public (call-foo (a uint))
+  (contract-call? .test-trait call-foo .impl-trait a)
+)
+`;
+
+  const callReadOnlyTestTraitWrapperCallFo = (
```
Comment (obycode) [inline tests/integration/epoch-2-2/wrap-trait-workaround3.spec.ts:63]:
```suggestion
  const callReadOnlyTestTraitWrapperCallFoo = (
```
(and its use below)

**Review by obycode — APPROVED** [review]
This looks great! Just noted some minor typos.

**Pull request merged**